src/org/yccheok/jstock/gui/charting/ChartJDialog.java

```diff
@@ -78,7 +85,7 @@
  *
  * @author  yccheok
  */
-public class ChartJDialog extends javax.swing.JDialog {
+public class ChartJDialog extends javax.swing.JFrame implements WindowListener {
```

Comment:
Yes, I am not thrilled with the multi-icon issue either but have a few ideas on how to deal with it. First, a new icon would be great but since JStock *is* a chart icon, I wasn't sure how to denote a chart on top of a chart icon, but if you give me an icon you like to represent charts happy to slot it in.

Another idea I had was dynamic icons eg take the stock icon and put in the ticker symbol in the icon so the user would know which chart they are actually going to. Not sure if that is actualyl do-able in Java, certainly do-able in win native app. My last idea would be to combine all charts into a tab-interfaced JFrame. That gets rid of the multi-icon problem doesn't do anything to solve one of my original objectives which is to see two items side-by-side. Ultimately what would be best there is to chart the two (or more) items on the single graph. All this is stuff I'l happy to work on if you give me some direction on which you prefer.